CRM/Case/BAO/Case.php

```diff
@@ -1212,9 +1212,20 @@ public static function getCaseActivity($caseID, &$params, $contactID, $context =
    *
    */
   public static function getRelatedContacts($caseID, $skipDetails = FALSE) {
+    $caseRoles = array();
+    if (!$skipDetails) {
+      $caseInfo = civicrm_api('Case', 'getsingle', array(
```

Comment:
So, the other thing here is... are you just accessing this via the api? This function is called from 3 places, 1 is the api & I feel like your changes are only meant for the benefit of the api. Obviously you need to change the query in this function - but I feel like the wrangling would be better in the api layer. 

It seems the api has the 'case_type_id.definition', already resolved (& the 2 other places in the code don't care about the extra data & will probably e-notice when they hit this line.

$role = CRM_Utils_Array::value($details['role'], $caseRoles);

As a general rule I support pushing stuff into the BAO layer for create & delete & actions, but doing handling for get actions in the API layer rather than in 'random BAO functions'

I'll have another go at the tests, thinking about it it makes sense that there must be able to quietly fail on an enotice or similar with civicrm_api but it noisily fails with civicrm-api3